music/src/core/melodies.ts

```diff
@@ -0,0 +1,142 @@
+/**
+ * Module containing functions for converting to and from quantized melodies.
+ *
+ * @license
+ * Copyright 2020 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Imports
+ */
+import {INoteSequence, NoteSequence} from '../protobuf/index';
+
+import * as sequences from './sequences';
+
+export const NO_EVENT = 0;
+export const NOTE_OFF = 1;
+const FIRST_PITCH = 2;
+
+/**
+ * Melody representation with note onsets, `NO_EVENT` (sustain) and `NOTE_OFF`.
```

Comment:
how is a note onset represented?

Reply:
and is this different than the Performance representation?

Reply:
The fact that I'm confused leads me to believe "melody" is too generic a name

Reply:
Added some doc to help clarify.